### Exploratory Data Analysis for The Vehicle Insurance Claim Fraud Detection System

Author: Hannah Lal
<br>Date: November 2023
<br>Email: itshannahlal@gmail.com

November 2023

#### Introduction

Insurance fraud is a prevalent and costly challenge for both policyholders and insurance companies. Fraudulent activities may encompass "false claims," "misrepresentation of information," or "organized fraud schemes." Detecting and preventing these activities is crucial for minimizing financial losses for issuers as well as safeguarding policyholders. Given its significant impact, insurance fraud detection has emerged as a prominent  research area in data science and machine learning. This capstone project aims to concentrate on identifying and preventing fraudulent or misleading insurance claims using the dataset available at Kaggle. The objective is to develop algorithms and models that can automatically detect suspicious activities through historical data analysis. 


Vehicle Insurance Fraud Detection is a vital component within the realm of insurance fraud prevention. This critical undertaking is centered on the identification and prevention of unauthorized or fraudulent claims related to vehicle insurance, representing a significant financial concern for all stakeholders involved. The challenge at hand is to devise methods that are both efficient and accurate in detecting these deceptive activities. Our focus in this capstone project is to address this challenge head-on, utilizing historical data accessible on the Kaggle platform. By immersing ourselves in the nuances of vehicle insurance fraud, we aspire to develop innovative solutions that enhance the resilience of insurance systems against the evolving landscape of fraudulent activities in the automotive insurance domain.


Stakeholders involved with issues related to Vehicle Insurance Fraud Detection are policyholders, insurance companies, and financial institutions. Policyholders stand to gain heightened security and protection against fraudulent claims, fortifying their confidence in the insurance process. Concurrently, insurance companies and financial institutions reap the benefits of diminished financial losses resulting from fraudulent activities, thereby bolstering customer trust and elevating their standing and reputation within the industry. This collaborative effort aims to create a more secure and trustworthy landscape in the realm of vehicle insurance, benefitting all entities involved in the insurance ecosystem.

//TODO: Explain about my project


### Data Dictionary
The vehicle insurance fraud dataset (fraud_oracle.csv) includes the following fields:

<!-- To be updated -->

| Features | Description |
|:------------|:------------|
| Month                   | Month of year the accident happened |
| WeekOfMonth             | Week number in a month the accident happened |
| DayOfWeek               | Day of the week the accident happened |
| Make                    | Brand of the vehicle |
| AccidentArea            | where the accident happened (Urban/Rural) |
| DayOfWeekClaimed        | Day of the week in which the claim was made |
| MonthClaimed            | The month in which the claim was made |
| WeekOfMonthClaimed      | The week number in the month during which the claim was made |
| Sex                     | Gender of the vehicle owner (Male/Female) |
| MaritalStatus           | Marital status of the vehicle owner |
| Age                     | Age of the vehicle owner |
| Fault                   | Fault of the accident:<br>Policy Holder: The fault of the accident is on Policy holder<br>Third Party: The fault of the accident is on the third party |
| PolicyType              | Indicates the type of coverage for the policy, specifying whether it covers liability, collision, or all perils for each vehicle type:<br>'Sport - Liability': Covers only liability for Sport vehicle category<br>'Sport - Collision': Covers only collision for Sport vehicle category<br>'Sedan - Liability': Covers only liability for Sedan vehicle category<br>'Utility - All Perils': Covers all perils for Utility vehicle category<br>'Sedan - All Perils': Covers all perils for Sedan vehicle category<br>'Sedan - Collision': Covers only collision for Sedan vehicle category<br>'Utility - Collision': Covers only collision for Utility vehicle category<br>'Utility - Liability': Covers only liability for Utility vehicle category<br>'Sport - All Perils': Covers all perils for Sport vehicle category |
| VehicleCategory         | Describes the category of the insured vehicle:<br>Sport: it is a sports car<br>Utility: it is a utility vehicle<br>Sedan: it is a sedan car |
| VehiclePrice            | The price range of the insured car:<br>'less than 20000'<br>'20000 to 29000'<br>'30000 to 39000'<br>'40000 to 59000'<br>'60000 to 69000'<br>'more than 69000' |
| PolicyNumber            | A unique identifier for each insurance policy |
| RepNumber               | Represents the identifier or code for the insurance representative or agent associated with the policy |
| Deductible              | Refers to the amount that the policyholder has to pay out of pocket before the insurance coverage takes effect |
| DriverRating            | Indicates the rating or risk associated with the driver insured under the policy (1 to 4) |
| Days_Policy_Accident    | The number of days since the last accident covered by the policy:<br>'none': Zero days since the last accident could mean the first time this car has been involved in an accident<br>'1 to 7': Last accident with the same car was in less than a week ago<br>'8 to 15': Last accident was any where between last two weeks and last week<br>'15 to 30': Last accident happened more than two weeks ago but less than a month ago<br>'more than 30': Last accident happened more than a month ago |
| Days_Policy_Claim       | The number of days since the last insurance claim was filed under the policy:<br>'none': Zero days since the last insurance claim was filed<br>'8 to 15': The last insurance claim was filed any time between at least a week ago and at most two weeks ago<br>'15 to 30': The last insurance claim was filed any time between at least 2 weeks ago and at most a month ago<br>'more than 30': The last insurance claim was filed any time at least a month ago |
| PastNumberOfClaims      | Represents the total number of claims filed by the policyholder in the past:<br>'none': No claims filed by the policyholder in the past<br>'1': The policyholder has a history of 1 claim being filed in the past<br>'2 to 4': The policyholder has a history of filing 2 to 4 claims in the past<br>'more than 4': The policyholder has a history of filing at least 5 claims (and more) in the past |
| AgeOfVehicle            | Age of the vehicle involved in the accident:<br>'new': The vehicle was new (less than a year)<br>'2 years': The vehicle was at most 2 years old<br>'3 years': The vehicle was at least 2 years old and at most 3 years old<br>'4 years': The vehicle was at least 3 years old and at most 4 years old<br>'5 years': The vehicle was at least 4 years old and at most 5 years old<br>'6 years': The vehicle was at least 5 years old and at most 6 years old<br>'7 years': The vehicle was at least 6 years old and at most 7 years old<br>'more than 7': The vehicle was at least 7 years old |
| AgeOfPolicyHolder       | The age range of the policyholder:<br>'16 to 17': The policyholder is between 16 and 17 years old<br>'18 to 20': The policyholder is between 18 and 20 years old<br>'21 to 25': The policyholder is between 21 and 25 years old<br>'26 to 30': The policyholder is between 26 and 30 years old<br>'31 to 35': The policyholder is between 31 and 35 years old<br>'36 to 40': The policyholder is between 36 and 40 years old<br>'41 to 50': The policyholder is between 41 and 50 years old<br>'51 to 65': The policyholder is between 51 and 65 years old<br>'over 65': The policyholder is at least over 65 years old |
| PoliceReportFiled       | Indicates whether a police report was filed in case of an accident or claim (Yes/No) |
| WitnessPresent          | Indicates whether any witnesses were present when the accident happened (Yes/No) |
| AgentType               | Describes the type or category of the insurance agent associated with the policy (External/Internal) |
| NumberOfSuppliments     | Indicates the number of additional supplements or endorsements to the policy:<br>'none': There were no additional supplements or endorsements to the policy<br>'1 to 2': There were 1 or 2 additional supplements or endorsements to the policy<br>'3 to 5': There were 3 to 5 additional supplements or endorsements to the policy<br>'more than 5': There were at least more than 5 additional supplements or endorsements to the policy |
| AddressChange_Claim     | Indicates whether there has been a change in the policyholder's address following a claim:<br>'no change': There has been no change in the policyholder's address following the claim<br>'under 6 months': The policyholder's address has been changed for under 6 months following the claim<br>'1 year': The policyholder's address has been changed for at least 6 months and at most a year following the claim<br>'2 to 3 years': The policyholder's address has been changed for at least 2 years and at most 3 years following the claim<br>'4 to 8 years': The policyholder's address has been changed for at least 4 years and at most 8 years following the claim |
| NumberOfCars            | Indicates the number of cars involved in the accident:<br>'1 vehicle': Only 1 vehicle was involved in the car accident<br>'2 vehicles': 2 vehicles were involved in the accident<br>'3 to 4': 3 to 4 cars were involved in the accident<br>'5 to 8': 5 to 8 cars were involved in the accident<br>'more than 8': More than 8 cars were involved in the accident |
| Year                    | The year the accident happened (1994, 1995, 1996) |
| BasePolicy              | Describes the foundational or base insurance policy that may include standard coverage:<br>'Liability': The base Insurance policy includes liability<br>'Collision': The base Insurance policy includes collision<br>'All Perils': The base Insurance policy includes all perils |

| Target Variable | Description |
|:------------|:------------|
| FraudFound_P   | Probability of a fraud being found |

### Table of contents
* Section 1: Initial Investigation of Data
* Section 2: Exploratory Data Analysis
* Section 3: Statistical Analysis
* Section 4: Preprocessing Data for Modeling
* Section 5: Conclusion and Insights

### Section 1: Initial Investigation of Data

In [6]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

# # filter warnings
# import warnings
# warnings.filterwarnings('ignore')

In [4]:
# read vehicle insurance fraud data:
df=pd.read_csv('C:/Users/itsha/Downloads/archive/fraud_oracle.csv')

In [7]:
# display leading 5 rows of data:
df.head()

Month  WeekOfMonth  DayOfWeek    Make AccidentArea DayOfWeekClaimed  \
0   Dec            5  Wednesday   Honda        Urban          Tuesday   
1   Jan            3  Wednesday   Honda        Urban           Monday   
2   Oct            5     Friday   Honda        Urban         Thursday   
3   Jun            2   Saturday  Toyota        Rural           Friday   
4   Jan            5     Monday   Honda        Urban          Tuesday   

  MonthClaimed  WeekOfMonthClaimed     Sex MaritalStatus  ...  AgeOfVehicle  \
0          Jan                   1  Female        Single  ...       3 years   
1          Jan                   4    Male        Single  ...       6 years   
2          Nov                   2    Male       Married  ...       7 years   
3          Jul                   1    Male       Married  ...   more than 7   
4          Feb                   2  Female        Single  ...       5 years   

  AgeOfPolicyHolder PoliceReportFiled WitnessPresent AgentType  \
0          26 to 30                No             No  External   
1          31 to 35               Yes             No  External   
2          41 to 50                No             No  External   
3          51 to 65               Yes             No  External   
4          31 to 35                No             No  External   

   NumberOfSuppliments  AddressChange_Claim  NumberOfCars  Year  BasePolicy  
0                 none               1 year        3 to 4  1994   Liability  
1                 none            no change     1 vehicle  1994   Collision  
2                 none            no change     1 vehicle  1994   Collision  
3          more than 5            no change     1 vehicle  1994   Liability  
4                 none            no change     1 vehicle  1994   Collision  

[5 rows x 33 columns]

In [11]:
# display last 5 rows of data:
df.tail()

Month  WeekOfMonth  DayOfWeek     Make AccidentArea DayOfWeekClaimed  \
15415   Nov            4     Friday   Toyota        Urban          Tuesday   
15416   Nov            5   Thursday  Pontiac        Urban           Friday   
15417   Nov            5   Thursday   Toyota        Rural           Friday   
15418   Dec            1     Monday   Toyota        Urban         Thursday   
15419   Dec            2  Wednesday   Toyota        Urban         Thursday   

      MonthClaimed  WeekOfMonthClaimed     Sex MaritalStatus  ...  \
15415          Nov                   5    Male       Married  ...   
15416          Dec                   1    Male       Married  ...   
15417          Dec                   1    Male        Single  ...   
15418          Dec                   2  Female       Married  ...   
15419          Dec                   3    Male        Single  ...   

       AgeOfVehicle AgeOfPolicyHolder PoliceReportFiled WitnessPresent  \
15415       6 years          31 to 35                No             No   
15416       6 years          31 to 35                No             No   
15417       5 years          26 to 30                No             No   
15418       2 years          31 to 35                No             No   
15419       5 years          26 to 30                No             No   

      AgentType  NumberOfSuppliments  AddressChange_Claim  NumberOfCars  Year  \
15415  External                 none            no change     1 vehicle  1996   
15416  External          more than 5            no change        3 to 4  1996   
15417  External               1 to 2            no change     1 vehicle  1996   
15418  External          more than 5            no change     1 vehicle  1996   
15419  External               1 to 2            no change     1 vehicle  1996   

       BasePolicy  
15415   Collision  
15416   Liability  
15417   Collision  
15418  All Perils  
15419   Collision  

[5 rows x 33 columns]

In [12]:
# display a sample 10 rows of data:
df.sample(10)

Month  WeekOfMonth  DayOfWeek       Make AccidentArea DayOfWeekClaimed  \
357     Mar            1     Friday     Toyota        Rural           Sunday   
5160    Jun            1     Friday  Chevrolet        Urban           Monday   
7405    Jun            4   Thursday    Pontiac        Urban           Monday   
541     Feb            4     Friday  Chevrolet        Urban           Friday   
2089    Sep            4  Wednesday    Pontiac        Urban         Thursday   
10993   Nov            1   Saturday    Pontiac        Urban          Tuesday   
9582    Mar            1     Friday       Ford        Rural        Wednesday   
7321    Jun            4   Thursday    Pontiac        Urban          Tuesday   
11430   Jun            2     Friday     Toyota        Rural         Thursday   
8759    Feb            4    Tuesday    Pontiac        Urban           Friday   

      MonthClaimed  WeekOfMonthClaimed     Sex MaritalStatus  ...  \
357            Mar                   1    Male       Married  ...   
5160           Jun                   3  Female       Married  ...   
7405           Jul                   1    Male        Single  ...   
541            Feb                   4    Male       Married  ...   
2089           Sep                   5    Male       Married  ...   
10993          Nov                   2    Male       Married  ...   
9582           Mar                   4    Male       Married  ...   
7321           Jun                   4    Male       Married  ...   
11430          Jun                   3    Male       Married  ...   
8759           Feb                   4    Male       Married  ...   

       AgeOfVehicle AgeOfPolicyHolder PoliceReportFiled WitnessPresent  \
357         7 years          36 to 40                No             No   
5160        6 years          31 to 35                No             No   
7405    more than 7          41 to 50                No             No   
541         7 years          41 to 50                No             No   
2089        7 years          36 to 40                No             No   
10993       7 years          36 to 40                No             No   
9582    more than 7          36 to 40                No             No   
7321    more than 7          41 to 50                No             No   
11430   more than 7          41 to 50                No             No   
8759    more than 7          51 to 65                No             No   

      AgentType  NumberOfSuppliments  AddressChange_Claim  NumberOfCars  Year  \
357    External               1 to 2         2 to 3 years    2 vehicles  1994   
5160   External                 none            no change     1 vehicle  1994   
7405   External                 none            no change     1 vehicle  1995   
541    External               1 to 2            no change     1 vehicle  1994   
2089   External                 none            no change     1 vehicle  1994   
10993  External               3 to 5            no change     1 vehicle  1995   
9582   External          more than 5            no change     1 vehicle  1995   
7321   External                 none            no change     1 vehicle  1995   
11430  External               1 to 2            no change     1 vehicle  1996   
8759   External               3 to 5            no change     1 vehicle  1995   

       BasePolicy  
357    All Perils  
5160    Liability  
7405    Collision  
541     Collision  
2089    Liability  
10993  All Perils  
9582    Collision  
7321   All Perils  
11430   Collision  
8759    Collision  

[10 rows x 33 columns]

Let's see how many rows and columns there are in the data set: 

In [9]:
df.shape

(15420, 33)

Let's see a summary of the data: 

In [15]:
# display summary of data:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15420 entries, 0 to 15419
Data columns (total 33 columns):
 #   Column                Non-Null Count  Dtype 
---  ------                --------------  ----- 
 0   Month                 15420 non-null  object
 1   WeekOfMonth           15420 non-null  int64 
 2   DayOfWeek             15420 non-null  object
 3   Make                  15420 non-null  object
 4   AccidentArea          15420 non-null  object
 5   DayOfWeekClaimed      15420 non-null  object
 6   MonthClaimed          15420 non-null  object
 7   WeekOfMonthClaimed    15420 non-null  int64 
 8   Sex                   15420 non-null  object
 9   MaritalStatus         15420 non-null  object
 10  Age                   15420 non-null  int64 
 11  Fault                 15420 non-null  object
 12  PolicyType            15420 non-null  object
 13  VehicleCategory       15420 non-null  object
 14  VehiclePrice          15420 non-null  object
 15  FraudFound_P          15420 non-null

We can see that:
- We have 15420 non-null values in all the columns. Therefore, no eliminating the need to fill in missing values as part of the preprocessing step in our analysis.
- There are 9 numerical columns, all being of type int64. Therefore, no floating point values in the columns.
- There are 24 categorical columns in the dataset. 

Now, let's see what the count, percentiles, mean, standard deviation, minimum and maximum values are like in each of the 9 numerical columns of the dataset:   

In [16]:
df.describe()

WeekOfMonth  WeekOfMonthClaimed           Age  FraudFound_P  \
count  15420.000000        15420.000000  15420.000000  15420.000000   
mean       2.788586            2.693969     39.855707      0.059857   
std        1.287585            1.259115     13.492377      0.237230   
min        1.000000            1.000000      0.000000      0.000000   
25%        2.000000            2.000000     31.000000      0.000000   
50%        3.000000            3.000000     38.000000      0.000000   
75%        4.000000            4.000000     48.000000      0.000000   
max        5.000000            5.000000     80.000000      1.000000   

       PolicyNumber     RepNumber    Deductible  DriverRating          Year  
count  15420.000000  15420.000000  15420.000000  15420.000000  15420.000000  
mean    7710.500000      8.483268    407.704280      2.487808   1994.866472  
std     4451.514911      4.599948     43.950998      1.119453      0.803313  
min        1.000000      1.000000    300.000000      1.000000   1994.000000  
25%     3855.750000      5.000000    400.000000      1.000000   1994.000000  
50%     7710.500000      8.000000    400.000000      2.000000   1995.000000  
75%    11565.250000     12.000000    400.000000      3.000000   1996.000000  
max    15420.000000     16.000000    700.000000      4.000000   1996.000000